Frends.PostgreSQL.ExecuteQuery/Frends.PostgreSQL.ExecuteQuery/ExecuteQuery.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Frends.PostgreSQL.ExecuteQuery.Definitions;
+using Newtonsoft.Json.Linq;
+using Npgsql;
+using System.Data;
+using System.ComponentModel;
+
+#pragma warning disable 1591
+
+namespace Frends.PostgreSQL.ExecuteQuery;
+
+public static class PostgreSQL
+{
+    /// <summary>
+    /// Query data using PostgreSQL. [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.PostgreSQL.ExecuteQuery)
+    /// </summary>
+    /// <param name="input">Query, parameters and connection string.</param>
+    /// <param name="options">Set timeout and isolation level.</param>
+    /// <param name="cancellationToken">Automatically generated and passed by Frends.</param>
+    /// <returns>Result of the query. JToken QueryResult</returns>
+    public static async Task<Result> ExecuteQuery([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        using (var conn = new NpgsqlConnection(input.ConnectionString))
+        {
+            await conn.OpenAsync(cancellationToken);
+            var transaction = conn.BeginTransaction(GetIsolationLevel(options.SqlTransactionIsolationLevel));
+            using (var cmd = new NpgsqlCommand(input.Query, conn))
+            {
+                cmd.CommandTimeout = options.CommandTimeoutSeconds;
+                cmd.Transaction = transaction;
+
+                // Add parameters to command, if any were given.
+                if (input.Parameters != null && input.Parameters.Length > 0)
+                {
+                    foreach (var parameter in input.Parameters)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        // Convert parameter.Value to DBNull.Value if it is set to null.
+                        if (parameter.Value == null)
+                        {
+                            cmd.Parameters.AddWithValue(parameter.Name, DBNull.Value);
+                        }
+                        else
+                        {
+                            cmd.Parameters.AddWithValue(parameter.Name, parameter.Value);
+                        }
+                    }
+                }
+
+                // Execute command.
+
+                if (input.Query.ToLower().Contains("select"))
+                {
+                    var reader = await cmd.ExecuteReaderAsync(cancellationToken);
+                    var result = reader.ToJson();
+                    conn.Close();
+                    return new Result(result);
+                }
+                else
+                {
+                    var rows = await cmd.ExecuteNonQueryAsync(cancellationToken);
+                    transaction.Commit();
+                    conn.Close();
+                    return new Result(JToken.FromObject(new { AffectedRows = rows }));
+                }
+            }
```

Comment:
You could add try catch block to catch all errors and add throw error on failure option like in all the other query tasks. You can also move the conn.Close() method to a finally block.